src/IIS/Manager/Base/BaseSiteManager.cs

```diff
@@ -1,624 +1,680 @@
 
-                if (settings.CertificateHash != null)
-                {
-                    site.Bindings[0].CertificateHash = settings.CertificateHash;
-                }
+            if (settings.CertificateHash != null)
+            {
+                site.Bindings[0].CertificateHash = settings.CertificateHash;
+            }
 
-                if (!String.IsNullOrEmpty(settings.CertificateStoreName))
-                {
-                    site.Bindings[0].CertificateStoreName = settings.CertificateStoreName;
-                }
+            if (!String.IsNullOrEmpty(settings.CertificateStoreName))
+            {
+                site.Bindings[0].CertificateStoreName = settings.CertificateStoreName;
+            }
 
-                site.ServerAutoStart = settings.ServerAutoStart;
-                site.ApplicationDefaults.ApplicationPoolName = settings.ApplicationPool.Name;
+            site.ServerAutoStart = settings.ServerAutoStart;
+            site.ApplicationDefaults.ApplicationPoolName = settings.ApplicationPool.Name;
 
 
 
-                //Security
-                this.SetAuthentication(settings);
-                this.SetAuthorization(settings);
+            //Security
+            this.SetAuthentication(settings);
+            this.SetAuthorization(settings);
 
-                return site;
-            }
+            return site;
+        }
+
+        protected const string webServerName = "webServer";
+        protected const string ftpServerName = "ftpServer";
+        /// <summary>
+        /// Sets the authentication settings for the site
+        /// </summary>
+        /// <param name="settings">The site settings</param>
+        protected void SetAuthentication(SiteSettings settings)
+        {
+            var authenticationSettings = settings.Authentication;
+            string server = "";
 
-            /// <summary>
-            /// Sets the authentication settings for the site
-            /// </summary>
-            /// <param name="settings">The site settings</param>
-            protected void SetAuthentication(SiteSettings settings)
+            if (settings is WebsiteSettings)
             {
-                if (settings.Authentication != null)
-                {
-                    //Get Type
-                    string server = "";
+                server = webServerName;
+            }
+            else
+            {
+                server = ftpServerName;
+            }
+            var locationPath = settings.Name;
 
-                    if (settings is WebsiteSettings)
-                    {
-                        server = "webServer";
-                    }
-                    else
-                    {
-                        server = "ftpServer";
-                    }
+            if (authenticationSettings != null)
+            {
+                //Authentication
+                SetSiteAuthentication(server, locationPath, authenticationSettings);
```

Comment:
Yes, I didn't want to break any public usage. And so when I divided that into two: SetSiteAuthentication, SetAppAuthentication then it would not break any code that was relying on SetAuthentication.